### PR TITLE
Overlapping shifts #132

### LINKED ISSUE #132: Implement Overlap Check for Selected Shifts on Sign Up for Shifts Page
**Is this task related to a feature or a bug? Please include relevant issue numbers below.**
Given the current functionality of the "Sign Up For Shifts" page, when a user selects multiple shifts, then the system should check for any overlapping shifts. If overlapping shifts are detected, the submit button should be disabled, and an icon (like an X) should be displayed to indicate the conflict. This feature is essential for preventing users from signing up for overlapping shifts.

**Describe the solution you'd like**
Implement a JavaScript function that performs the following:

- Loop through the currently selected shifts on the "Sign Up For Shifts" page.
- Check for overlapping time slots among the selected shifts.
- If any overlaps are found, disable the submit button.
- Display an icon (e.g., an X) to indicate that overlapping shifts have been selected.

**Acceptance Criteria:**

- [ ] The JavaScript function should accurately detect overlapping shifts based on their start and end times.
- [ ] The submit button should be disabled if overlaps are found.
- [ ] An icon should be displayed to indicate overlapping shifts.
- [ ] The functionality should work seamlessly on the "Sign Up For Shifts" page.

### PR DESCRIPTION
**Pull request by amalrzv**
Fixes #132 

**What was changed?**
Implemented an overlapping shifts section that indicates whether shifts overlap in the Shelters.js file

*Here, describe what part of the application you changed (e.g. login page, database, etc.). If possible, mention what specific components were added, removed, or modified.*

The current shifts component was changed where you sign up for shifts. 

**Why was it changed?**
We needed to find a way to indicate to the volunteer when the shifts they selected are overlapping times. 

*Here, describe the issue that you are fixing with this code, and why your code fixes it.*

Before, we were not indicating with a message that says that shifts are overlapping and allowing volunteers to submit shifts that were overlapping (even with the success and failure detection). Now, if a volunteer adds 2 shifts that are overlapping, red text appears on the overlapping shifts and the submit button is disabled. 

**How was it changed?**

*Here, get into detail about what files you modified, and talk about the most important lines in regards to fixing the issue.*

I created a new file to hold the current shifts component. There, I passed the Shift List component to display selected shifts and handle removal actions. I kept the overall overlap check at the component level to control the "Submit Shifts" button's disabled state effectively. The shift submission was also on the Shelters.js and I added the currentselection component which is displaying the current selection of shifts, removing shifts, and checking for overlaps before submission. 

**Screenshots that show the changes (if applicable):**
Screenshot 2024-10-20 at 6 31 01 PM

Screenshot 2024-10-20 at 6 31 23 PM

### CODE REVIEW

**Review by cbiddle3 — COMMENTED** [review]
Already discussed in class but moving this logic to the current selection part of the website

File: client_app/src/components/volunteer/ConfirmationPage.js [hunk 1]
```diff
@@ -3,12 +3,28 @@ import { Link } from "react-router-dom";
...
+import CancelIcon from "@mui/icons-material/Cancel";
 import CheckCircleIcon from "@mui/icons-material/CheckCircle";
 
 const handleReload = () => {
   window.location.reload();
 };
 
+
+const checkForOverlap = (shift, otherShifts) => {
+  return otherShifts.some(otherShift => {
```
Comment (cbiddle3) [inline client_app/src/components/volunteer/ConfirmationPage.js:15]: you should be able to just directly compare the shift.start_time and shift.end_time

File: client_app/src/components/volunteer/ConfirmationPage.js [hunk 2]
```diff
@@ -69,6 +89,18 @@ const ConfirmationPage = ({ selectedShifts, shiftStatusList }) => {
...
                   </p>
                 </td>
+                <td>
+                  {/* Check and display if the shift has an overlap */}
+                  {checkForOverlap(shift, selectedShifts.filter((_, i) => i !== index)) ? (
+                    <IconButton>
+                      <CancelIcon className="overlap-icon" style={{ color: "red" }} />
+                    </IconButton>
+                  ) : (
+                    <IconButton>
```
Comment (cbiddle3) [inline client_app/src/components/volunteer/ConfirmationPage.js:99]: Discussed in class with error message and tooltip

**Review by cbiddle3 — CHANGES_REQUESTED** [review]
Just make the changes above moving logic to current selection

**Review by Brijitha1609 — COMMENTED** [review]
Great job! To further enhance the user experience, it might be helpful to include a visual indication, such as an icon (like an "X"), to alert users when overlapping shifts have been selected. This additional feedback can guide users in making more informed selections.

**Review by Brijitha1609 — APPROVED** [review]
Great job implementing the overlap detection functionality and ensuring that it works as expected. You've provided clear visual feedback to the user when overlapping shifts are detected, which is essential for usability. The approach you've taken to manage overlapping shifts is logical and well-structured.

**Pull request merged**